Flaky test: TestManagerEncryptsDecryptsRootKeyMaterial

This seems like it could have been affected by #2128.

```
--- FAIL: TestManagerEncryptsDecryptsRootKeyMaterial (0.37s)
	Error Trace:	manager_test.go:668
	Error:		Received unexpected error raft: failed to process the request: node lost leader status
			github.com/docker/swarmkit/manager/state/raft.init
				/home/ubuntu/.go_workspace/src/github.com/docker/swarmkit/manager/state/raft/raft.go:51
			github.com/docker/swarmkit/manager.init
				/home/ubuntu/.go_workspace/src/github.com/docker/swarmkit/manager/manager_test.go:679
			main.init
				github.com/docker/swarmkit/manager/_test/_testmain.go:134
			runtime.main
				/usr/local/go/src/runtime/proc.go:172
			runtime.goexit
				/usr/local/go/src/runtime/asm_amd64.s:2086
```

--- a/manager/manager_test.go
+++ b/manager/manager_test.go
@@ -630,6 +630,11 @@ func TestManagerEncryptsDecryptsRootKeyMaterial(t *testing.T) {
 	// wait for the key to be decrypted in the raft store
 	pollDecrypted := func() error {
 		return testutils.PollFunc(nil, func() error {
+			// wait until we are leader first, because otherwise the raft node could still be catching
+			// up on all the logs on disk and hence not have processed the "encrypt CA key" log yet
+			if !m.raftNode.IsLeader() {
+				return fmt.Errorf("node is not leader yet")
+			}
 			return m.raftNode.MemoryStore().Update(func(tx store.Tx) error {
 				cluster := store.GetCluster(tx, clusterID)
 				if cluster == nil {
@@ -648,7 +653,9 @@ func TestManagerEncryptsDecryptsRootKeyMaterial(t *testing.T) {
 	}
 	require.NoError(t, pollDecrypted())
 
-	// update the key to that can be "decrypted" with both "" and "kek" as the password
+	// update the key to that can be "decrypted" with both "" and "kek" as the password.  This
+	// doesn't actually match the root CA certificate, and hence the security config can't be
+	// updated, but we're just checking that the CA key is decrypted.
 	require.NoError(t, m.raftNode.MemoryStore().Update(func(tx store.Tx) error {
 		cluster := store.GetCluster(tx, clusterID)
 		if cluster == nil {

